examples/flutter_gallery/lib/gallery/backdrop.dart

```diff
@@ -0,0 +1,329 @@
+        style: theme.primaryTextTheme.title,
+        child: new SizedBox(
+          height: _kBackAppBarHeight,
+          child: new Row(children: children),
+        ),
+      ),
+    );
+  }
+}
+
+class Backdrop extends StatefulWidget {
+  const Backdrop({
+    this.frontAction,
+    this.frontTitle,
+    this.frontHeading,
+    this.frontLayer,
+    this.backTitle,
+    this.backLayer,
+  });
+
+  final Widget frontAction;
+  final Widget frontTitle;
+  final Widget frontLayer;
+  final Widget frontHeading;
+  final Widget backTitle;
+  final Widget backLayer;
+
+  @override
+  _BackdropState createState() => new _BackdropState();
+}
+
+class _BackdropState extends State<Backdrop> with SingleTickerProviderStateMixin {
+  final GlobalKey _backdropKey = new GlobalKey(debugLabel: 'Backdrop');
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = new AnimationController(
+      duration: const Duration(milliseconds: 300),
+      value: 1.0,
+      vsync: this,
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  double get _backdropHeight {
+    final RenderBox renderBox = _backdropKey.currentContext.findRenderObject();
+    return math.max(0.0, renderBox.size.height - _kBackAppBarHeight - _kFrontClosedHeight);
+  }
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    _controller.value -= details.primaryDelta / (_backdropHeight ?? details.primaryDelta);
+  }
+
+  void _handleDragEnd(DragEndDetails details) {
+    if (_controller.isAnimating || _controller.status == AnimationStatus.completed)
+      return;
+
+    final double flingVelocity = details.velocity.pixelsPerSecond.dy / _backdropHeight;
+    if (flingVelocity < 0.0)
+      _controller.fling(velocity: math.max(2.0, -flingVelocity));
+    else if (flingVelocity > 0.0)
+      _controller.fling(velocity: math.min(-2.0, -flingVelocity));
+    else
+      _controller.fling(velocity: _controller.value < 0.5 ? -2.0 : 2.0);
+  }
+
+  void _toggleFrontLayer() {
+    final AnimationStatus status = _controller.status;
+    final bool isOpen = status == AnimationStatus.completed || status == AnimationStatus.forward;
+    _controller.fling(velocity: isOpen ? -2.0 : 2.0);
+  }
+
+  Widget _buildStack(BuildContext context, BoxConstraints constraints) {
+    final Animation<RelativeRect> frontRelativeRect = new RelativeRectTween(
+      begin: new RelativeRect.fromLTRB(0.0, constraints.biggest.height - _kFrontClosedHeight, 0.0, 0.0),
+      end: const RelativeRect.fromLTRB(0.0, _kBackAppBarHeight, 0.0, 0.0),
+    ).animate(_controller);
+
+    return new Stack(
+      key: _backdropKey,
+      children: <Widget>[
+        new Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: <Widget>[
+            // Back layer
+            new _BackAppBar(
+              leading: widget.frontAction,
+              title: new _CrossFadeTransition(
```

Comment:
Does reusing `AnimatedCrossFade` not work?

Reply:
AnimatedCrossFade is an implicit animation, this one is effectively driven by the front layer's scroll offset.